### ISSUE TITLE
[Android][iOS] Separate Interpreter Lib Asserts When Used

### ISSUE THREAD

**Issue by steveisok** [dotnet/runtime#44804]
When forcing and initializing the interpreter lib to be used:

```
mono_jit_set_aot_mode(MONO_AOT_MODE_INTERP_ONLY);
mono_ee_interp_init ("");
```

The following assertions get hit:

**Android**

```
D DOTNET  : assembly_preload_hook: System.ObjectModel  /data/user/0/net.dot.System.Buffers.Tests/files
D DOTNET  : assembly_preload_hook: System.Text.RegularExpressions  /data/user/0/net.dot.System.Buffers.Tests/files
D DOTNET  : ((null) error) * Assertion: should not be reached at src/mono/mono/mini/tramp-amd64.c:1131
```

**iOS**
```
2020-11-18 19:23:37.254713-0500 System.Buffers.Tests[86762:4399725] 
((null) error) * Assertion: should not be reached at src/mono/mono/mini/tramp-amd64.c:1131
```

https://github.com/dotnet/runtime/blob/9c1fde0868d9d8f193bdf838519e7383123ebfe7/src/mono/mono/mini/tramp-amd64.c#L1131

**Comment by SamMonoRT** [comment]
@steveisok - is this still relevant ?

**Comment by steveisok** [comment]
if we want to support interp as a separate library, probably.

**Comment by BrzVlad** [comment]
This is some sort of issue in the build configuration. According to the assertion, mono is built with `DISABLE_INTERPRETER` which is incorrect. It would seem that this is added here  https://github.com/dotnet/runtime/blob/main/src/mono/CMakeLists.txt#L93 and I have no idea why. @vargaz Any ideas ? Seems like you added this code last, but I'm not sure where it originates from.

**Comment by vargaz** [comment]
DISABLE_INTERPRETER right now means that libmonosgen doesn't have interp support, but its in the separate libmono-interp. Its a bit of a messy situation.

**Comment by BrzVlad** [comment]
The interpreter trampoline generation is not part of the interpreter library, it is in mini. So if we ever want to use the interpreter, then these shouldn't really be disabled. It is also counter intuitive to me that the runtime has disable interpreter but we can still use it.

**Comment by vargaz** [comment]
Yeah, we need a separate define for the separate interpreter library, not DISABLE_INTERPRETER.

**Comment by BrzVlad** [comment]
I think https://github.com/BrzVlad/runtime/commit/ed19f6eed681293d9891fd6b6a75dc579d98be27 is a fix for this. @steveisok Could you give it a try ?

**Comment by SamMonoRT** [comment]
@steveisok @BrzVlad - does Vlad's commit fix the issue ?

**Comment by steveisok** [comment]
I'll give it a try and let you know

**Comment by SamMonoRT** [comment]
> I'll give it a try and let you know

Any update on this Steve ?

**Comment by SamMonoRT** [comment]
Moving this to Future milestone.
